include/translatable-object.php

```diff
@@ -246,53 +246,110 @@ public function delete_language( $id ) {
+		}
+		_prime_term_caches( $all_term_ids, false );
+
+		$terms = array();
+		foreach ( $cached_values as $object_id => $term_ids ) {
+			if ( ! empty( $term_ids ) ) {
+				$term_id             = reset( $term_ids ); // There is only one term for language or translation groups.
+				$term                = get_term( $term_id );
+				$terms[ $object_id ] = $term instanceof WP_Term ? $term : null;
+			}
 		}
 
-		$term = get_object_term_cache( $id, $taxonomy );
+		return $terms;
+	}
+
+	/**
+	 * Caches all object-relationship terms.
+	 *
+	 * @since 3.8
+	 *
+	 * @param int[] $object_ids Array of object IDs to retrieve terms for.
+	 *
+	 * @return void
+	 */
+	private function update_object_term_cache( array $object_ids ): void {
+		$non_cached_ids = array();
+		foreach ( $this->tax_to_cache as $taxonomy ) {
+			$non_cached_ids = array_merge( $non_cached_ids, _get_non_cached_ids( $object_ids, "{$taxonomy}_relationships" ) );
+		}
 
-		if ( is_array( $term ) ) {
-			return ! empty( $term ) ? reset( $term ) : false;
+		if ( empty( $non_cached_ids ) ) {
+			return;
 		}
 
-		// Query terms.
-		$terms        = array();
-		$term         = false;
-		$object_terms = wp_get_object_terms( $id, $this->tax_to_cache, array( 'update_term_meta_cache' => false ) );
+		$terms = wp_get_object_terms(
+			array_unique( $non_cached_ids ),
+			$this->tax_to_cache,
+			array(
+				'fields'                 => 'all_with_object_id',
+				'update_term_meta_cache' => false,
+			)
+		);
 
-		if ( is_array( $object_terms ) ) {
-			foreach ( $object_terms as $t ) {
-				$terms[ $t->taxonomy ] = $t;
-				if ( $t->taxonomy === $taxonomy ) {
-					$term = $t;
-				}
+		if ( ! is_array( $terms ) ) {
+			return;
+		}
+
+		$object_terms = array();
+		foreach ( $terms as $term ) {
+			if ( ! isset( $term->object_id ) ) {
+				continue;
 			}
+
+			$object_terms[ $term->taxonomy ][ $term->object_id ][] = $term->term_id;
 		}
 
-		foreach ( $this->tax_to_cache as $tax ) {
-			if ( empty( $terms[ $tax ] ) ) {
-				$to_cache = array();
-			} else {
-				$to_cache = array( $terms[ $tax ]->term_id );
+		foreach ( $non_cached_ids as $id ) {
+			foreach ( $this->tax_to_cache as $taxonomy ) {
+				if ( ! isset( $object_terms[ $taxonomy ][ $id ] ) ) {
+					$object_terms[ $taxonomy ][ $id ] = array();
+				}
 			}
+		}
 
-			wp_cache_add( $id, $to_cache, "{$tax}_relationships" );
+		foreach ( $object_terms as $tax => $data ) {
+			wp_cache_add_multiple( $data, "{$tax}_relationships" );
 		}
+	}
 
-		return $term;
+	/**
+	 * Wraps `wp_get_object_terms()` to cache it and return only one object.
+	 *
+	 * @since 1.2
+	 *
+	 * @param int    $id       Object ID.
+	 * @param string $taxonomy Polylang taxonomy depending if we are looking for a post (or term, or else) language.
+	 * @return WP_Term|null The term associated to the object in the requested taxonomy if it exists, `false` otherwise.
+	 */
+	public function get_object_term( $id, $taxonomy ) {
```

Comment:
To improve the readability, I suggest to rename object ids variables to `$object_id` everywhere in the new methods.

Reply:
Your comment is on an old method. Do you mean to modify also here ?

New methods concerned are in `PLL_Translated_Object` class.
Done in 7c75eed6b93d86e6090018ef496f0a568210d793

Reply:
Done in a45232313307db7f09194b9a64b6a42ef376a328